tests/unit/test_placeholders_with_steps.py

```diff
@@ -387,3 +409,53 @@ def test_schema_validation_for_step_input():
                 "ParamB": "SampleValueB"
             }
         )
+
+def test_step_result_placeholder():
+    step_result = StepResult()
```

Comment:
Probably want to add a successful test case *with* a schema too

Reply:
added a test for successful schema